scipy/optimize/_remove_redundancy.py

```diff
@@ -230,6 +231,37 @@ def _remove_redundancy_dense(A, rhs):
     return A_orig[keep, :], rhs[keep], status, message
 
 
+class LUSolver:
+    def __init__(self, m):
+        I = eye(m, format='csc')
+        self.fl = splu(I, permc_spec="NATURAL")
+        self.fu = splu(I, permc_spec="NATURAL")
```

Comment:
Separate SuperLU objects representing the `L` matrix and the `U` matrix because I need to be able to update the `L` and `U` factors. SuperLU objects don't allow me to change their `L` and `U` attributes. I get around that perfoming updates on `L` and `U` sparse matrices and creating a separate `SuperLU` object from each of them.